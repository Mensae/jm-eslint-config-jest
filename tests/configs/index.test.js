/* eslint-env jest */

const jestConfig = require('../../configs/index.js');

describe('eslint-config-jest', () => {
  it('should match the snapshot', () => {
    expect(jestConfig).toMatchSnapshot();
  });
});
