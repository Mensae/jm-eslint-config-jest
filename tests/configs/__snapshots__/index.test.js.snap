// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-jest should match the snapshot 1`] = `
{
  "overrides": [
    {
      "env": {
        "jest/globals": true,
      },
      "files": [
        "**/__tests__/**",
        "*.test.*",
        "*.spec.*",
      ],
      "rules": {
        "@getify/proper-arrows/name": "off",
        "@getify/proper-arrows/this": "off",
        "@getify/proper-arrows/where": "off",
        "jest-extended/prefer-to-be-array": "warn",
        "jest-extended/prefer-to-be-false": "warn",
        "jest-extended/prefer-to-be-object": "warn",
        "jest-extended/prefer-to-be-true": "warn",
        "jest-extended/prefer-to-have-been-called-once": "warn",
        "jest/expect-expect": "warn",
        "jest/no-alias-methods": "error",
        "jest/no-commented-out-tests": "warn",
        "jest/no-conditional-expect": "error",
        "jest/no-deprecated-functions": "error",
        "jest/no-disabled-tests": "warn",
        "jest/no-done-callback": "error",
        "jest/no-export": "error",
        "jest/no-focused-tests": "error",
        "jest/no-identical-title": "error",
        "jest/no-interpolation-in-snapshots": "error",
        "jest/no-jasmine-globals": "error",
        "jest/no-mocks-import": "error",
        "jest/no-restricted-matchers": [
          "warn",
          {
            "toBeDefined": "Use a more explicit matcher. See https://docs.gitlab.com/ee/development/testing_guide/frontend_testing.html#tricky-tobedefined-matcher",
            "toBeFalsy": "Use more explicit matchers like \`.toBe(false)\`, \`.toBe(0)\`, or \`.toBeUndefined()\`. See https://docs.gitlab.com/ee/development/testing_guide/frontend_testing.html#avoid-using-tobetruthy-or-tobefalsy",
            "toBeTruthy": "Use more explicit matchers like \`.toBe(true)\` or \`.toBeGreaterThan(0)\`. See https://docs.gitlab.com/ee/development/testing_guide/frontend_testing.html#avoid-using-tobetruthy-or-tobefalsy",
          },
        ],
        "jest/no-standalone-expect": "error",
        "jest/no-test-prefixes": "error",
        "jest/valid-describe-callback": "error",
        "jest/valid-expect": [
          "error",
          {
            "alwaysAwait": true,
          },
        ],
        "jest/valid-expect-in-promise": "error",
        "jest/valid-title": "error",
      },
    },
    {
      "env": {
        "browser": false,
        "node": true,
      },
      "files": [
        "jest.config.js",
        "jest.config.cjs",
      ],
      "parserOptions": {
        "sourceType": "script",
      },
    },
  ],
  "plugins": [
    "import",
    "jest",
    "jest-extended",
    "unicorn",
  ],
  "settings": {
    "jest": {
      "version": "29.5.0",
    },
  },
}
`;
